torsiondrive/extra_constraints.py

```diff
@@ -84,12 +84,18 @@ def check_conflict_constraits(constraints_dict, dihedral_idxs):
     Utility function to check if any extra constraints in constraints_dict is conflict with the scanning dihedrals
     """
     distinct_dihedrals = set(tuple(min(d, d[::-1])) for d in dihedral_idxs)
+    distinct_dihedral_centers = set(tuple(min(d[1:3], d[2:0:-1])) for d in dihedral_idxs)
     for constraits_list in constraints_dict.values():
         for spec_dict in constraits_list:
             ctype, indices = spec_dict['type'], spec_dict['indices']
-            if ctype == 'dihedral' and tuple(min(indices, indices[::-1])) in distinct_dihedrals:
-                raise ValueError(f"Conflict dihedral constraints found in:\n{spec_dict}\n with {dihedral_idxs}")
-            if ctype == 'xyz':
+            if ctype == 'dihedral':
+                if tuple(min(indices, indices[::-1])) in distinct_dihedrals:
+                    # the same dihedral appears in extra_constraints
+                    raise ValueError(f"Conflict dihedral constraints found in:\n{spec_dict}\n with {dihedral_idxs}")
+                elif tuple(min(indices[1:3], indices[2:0:-1])) in distinct_dihedral_centers:
```

Comment:
Thanks, this looks good. Please correct spelling `check_conflict_constraits` to `check_conflict_constraints`

Reply:
Oops, thanks for pointing out